frontend/src/forms/subforms/BuildingForm.tsx

```diff
@@ -78,21 +80,11 @@ const BuildingForm = <T extends any>(props: BuildingProps & FormikProps<T>) => {
     return [props.nameSpace ?? '', `${props.index ?? ''}`, name].filter(x => x).join('.');
   };
 
-  //set the lat/lon of this building, but only if the building lat/lon hasn't been touched and has no value.
-  if (
-    !getIn(props.touched, withNameSpace('latitude')) &&
-    !getIn(props.values, withNameSpace('latitude')) &&
-    props.values.latitude &&
-    props.touched.latitude
-  ) {
+  //set the lat/lon of this building, but only if the building lat/lon has no value.
+  if (!getIn(props.values, withNameSpace('latitude')) && props.values.latitude) {
```

Comment:
Was never reaching these statements when it was checking if the values were not touched. Now it fills the lat long as expected and can update it afterwards if it is not where desired.